### ISSUE TITLE
[a11y]: [UI Shell] [SideNav with Divider] - Screen reader misinterprets two lists as one and counts the divider as an item

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16888]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

v1.60.3

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

<img width="295" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/61d00fe2-ad50-4c48-a28d-27de4f804234">

Visually, there are two distinct lists separated by a divider. The first list contains three items, and the second list contains two items. However, the screen reader announces these as a single list and incorrectly counts the divider as an additional item.

Two different list should be defined and screen reader should announce the list items count along with the list correctly.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--fixed-side-nav-w-divider

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
